File: SteamKit/Source/Categories/SteamUser+Local.m
//
//  SteamUser+Local.m
//  SteamKit
//
//  Created by Serhii Londar on 10/24/16.
//  Copyright © 2016 SLON. All rights reserved.
//

#import "SteamUser+Local.h"
#import <RMMapper.h>

@implementation SteamUser (Local)

+ (SteamUser *)load {
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    NSData *encodedObject = [defaults objectForKey:@"SteamUser"];
    SteamUser *object = [NSKeyedUnarchiver unarchiveObjectWithData:encodedObject];
    return object;
}

- (void)save {
    NSData *encodedObject = [NSKeyedArchiver archivedDataWithRootObject:self];
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    [defaults setObject:encodedObject forKey:@"SteamUser"];
    [defaults synchronize];
}


- (void)encodeWithCoder:(NSCoder *)encoder {
    //Encode properties, other class variables, etc
    [encoder encodeObject:self.steamID64 forKey:@"steamID64"];
    [encoder encodeObject:self.steamID32 forKey:@"steamID32"];
    [encoder encodeObject:self.steamVanityID forKey:@"steamVanityID"];
}

- (id)initWithCoder:(NSCoder *)decoder {
    if((self = [super init])) {
        //decode properties, other class vars
        self.steamID64 = [decoder decodeObjectForKey:@"steamID64"];
        self.steamID32 = [decoder decodeObjectForKey:@"steamID32"];
        self.steamVanityID = [decoder decodeObjectForKey:@"steamVanityID"];
    }
    return self;
}

@end

File: SteamLogin/Source/Categories/SteamUser+Local.swift
//
//  SteamUser+Local.swift
//  SteamKit
//
//  Created by Serhii Londar on 12/15/17.
//  Copyright © 2017 slon. All rights reserved.
//

import Foundation

//extension SteamUser: NSCoding {
//    static func loadUser() -> SteamUser? {
//        let defaults: UserDefaults? = UserDefaults.standard
//        let encodedObject = defaults?.object(forKey: "SteamUser") as? Data
//        let object = NSKeyedUnarchiver.unarchiveObject(with: encodedObject ?? Data()) as? SteamUser
//        return object
//    }
//
//    func save() {
//        let encodedObject = NSKeyedArchiver.archivedData(withRootObject: self)
//        let defaults: UserDefaults? = UserDefaults.standard
//        defaults?.set(encodedObject, forKey: "SteamUser")
//        defaults?.synchronize()
//    }
//
//    func encode(with encoder: NSCoder) {
//        encoder.encode(steamID64, forKey: "steamID64")
//        encoder.encode(steamID32, forKey: "steamID32")
//        encoder.encode(steamVanityID, forKey: "steamVanityID")
//    }
//
//    convenience init?(coder aDecoder: NSCoder) {
//        super.init(aDecoder)
//        steamID64 = aDecoder.decodeObject(forKey: "steamID64") as? String
//        steamID32 = aDecoder.decodeObject(forKey: "steamID32") as? String
//        steamVanityID = aDecoder.decodeObject(forKey: "steamVanityID") as? String
//    }
//}

